cbmc/proofs/mqtt_state.c

```diff
@@ -402,57 +431,63 @@ bool valid_IotMqttSubscriptionListElt( const _mqttSubscription_t *pElt )
     // MAX is one greater than the maximum length
     pElt->topicFilterLength < TOPIC_LENGTH_MAX &&
 #endif
-    pElt->topicFilterLength < CBMC_MAX_OBJECT_SIZE - sizeof( *pElt ) &&
-    __CPROVER_r_ok( pElt->pTopicFilter, pElt->topicFilterLength ) &&
-    __CPROVER_w_ok( pElt->pTopicFilter, pElt->topicFilterLength );
+    pElt->topicFilterLength < CBMC_MAX_OBJECT_SIZE - sizeof( *pElt );
 }
 
 // Subscription linked lists
 
 IotListDouble_t *allocate_IotMqttSubscriptionList( IotListDouble_t *pSub,
 						   size_t length )
 {
-  if ( pSub == NULL ) pSub = malloc_can_fail( sizeof( *pSub ) );
-  if ( pSub == NULL ) return NULL;
-
   // Allocate lists of length L <= 3 (MAX = L+1)
-  __CPROVER_assert(SUBSCRIPTION_COUNT_MAX <= 3+1,
-		   "Subscription list bound is too big");
   __CPROVER_assert(length < SUBSCRIPTION_COUNT_MAX,
-		   "Subscription list requested is too long");
+		   "Subscription list length max is greater than MAX");
+  __CPROVER_assert(length <= 3,
+		   "Subscription list length max is greater than 3");
+
+  if ( pSub == NULL ) pSub = malloc_can_fail( sizeof( *pSub ) );
+  if ( pSub == NULL ) return NULL;
 
   IotListDouble_Create( pSub );
 
   size_t num_elts;
   __CPROVER_assume(num_elts <= length);
 
-  if (1 <= num_elts)
-    {
-      _mqttSubscription_t *pElt = allocate_IotMqttSubscriptionListElt( NULL );
+  _mqttSubscription_t *pElt;
+  switch (num_elts) {
```

Comment:
Same comment as in ```allocate_IotMqttOperationList```.

Reply:
Same response as in `allocate_IotMqttOperationList`.